### PR TITLE
Add source generator to auto-create model package wrapper

### LINKED ISSUE #8: Source generator to auto-create model package wrapper class
## Summary

Create a C# source generator that reads the embedded `model-manifest.json` at compile time and auto-generates the boilerplate wrapper class (e.g., `MiniLMModel.cs`), reducing model package authoring to just a manifest file and a `.csproj`.

## Problem

Today, creating a model package requires writing a wrapper class like `MiniLMModel.cs` that:
- Creates a `Lazy<ModelPackage>` from the embedded manifest
- Exposes `EnsureModelAsync`, `GetModelInfoAsync`, `VerifyModelAsync`
- Optionally wraps the model in an inference API (e.g., `CreateEmbeddingGeneratorAsync`)

The first three are pure boilerplate  identical across every model package, differing only in the class name and assembly reference. A source generator can eliminate this.

### 1. Create a source generator project

```
src/
  ModelPackages.Generators/
    ModelPackages.Generators.csproj    targets netstandard2.0 (required for generators)
    ModelPackageGenerator.cs
```

### 2. Trigger: embedded resource named `model-manifest.json`

The generator scans the compilation's additional files and embedded resources for a file named `model-manifest.json`. If found, it generates a partial static class.

### 3. Generated code

For a project with a manifest for `sentence-transformers/all-MiniLM-L6-v2`, the generator produces:

`csharp
// <auto-generated />
namespace SampleModelPackage.Onnx;

public static partial class ModelPackageApi
{
    private static readonly Lazy<ModelPackages.ModelPackage> _package = new(() =>
        ModelPackages.ModelPackage.FromManifestResource(typeof(ModelPackageApi).Assembly));

    public static Task<string> EnsureModelAsync(
        ModelPackages.ModelOptions? options = null, CancellationToken ct = default)
        => _package.Value.EnsureModelAsync(options, ct);

    public static Task<ModelPackages.ModelInfo> GetModelInfoAsync(
        ModelPackages.ModelOptions? options = null, CancellationToken ct = default)
        => _package.Value.GetModelInfoAsync(options, ct);

    public static Task VerifyModelAsync(
        ModelPackages.ModelOptions? options = null, CancellationToken ct = default)
        => _package.Value.VerifyModelAsync(options, ct);

    public static void ClearCache(
        ModelPackages.ModelOptions? options = null)
        => _package.Value.ClearCache(options);
}
`

### 4. The class is `partial`

Model package authors can extend it with inference-specific methods:

`csharp
public static partial class ModelPackageApi
{
    public static async Task<IEmbeddingGenerator<string, Embedding<float>>>
        CreateEmbeddingGeneratorAsync(ModelOptions? options = null, CancellationToken ct = default)
    {
        var modelPath = await EnsureModelAsync(options, ct);
        // ... build inference pipeline ...
    }
}
`

### 5. Optional: customize class name via attribute or MSBuild property

`xml
<PropertyGroup>
  <ModelPackageClassName>MiniLMModel</ModelPackageClassName>
</PropertyGroup>
`

Or via an assembly attribute:

`csharp
[assembly: ModelPackageClass(""MiniLMModel"")]
`

## Files to Create

- `src/ModelPackages.Generators/ModelPackages.Generators.csproj`  Source generator project (netstandard2.0, analyzer output)
- `src/ModelPackages.Generators/ModelPackageGenerator.cs`  The `IIncrementalGenerator` implementation
- `src/ModelPackages.Generators/ManifestParser.cs`  Lightweight JSON parser (can't use STJ source generators inside a generator)

## Files to Modify

- `ModelPackages.slnx`  Add generator project
- `src/ModelPackages/ModelPackages.csproj`  Optionally reference the generator as an analyzer so consuming projects get it automatically

## Design Considerations

- Source generators **cannot** read embedded resources at compile time (they're embedded at build time, not available during generation). The generator must read the manifest as an `AdditionalFile`:
  `xml
  <ItemGroup>
    <AdditionalFiles Include=""model-manifest.json"" />
  </ItemGroup>
  `
  This means the `.csproj` needs both `EmbeddedResource` (for runtime) and `AdditionalFiles` (for source gen).

- The generator should produce clean, readable code with `auto-generated` comment.
- The generator should handle edge cases: missing manifest, malformed JSON, missing model ID.

## Testing Suggestions

- Unit test: generator produces valid C# for a well-formed manifest
- Unit test: no manifest file  no generated code (not an error)
- Unit test: malformed manifest  diagnostic warning, no generated code
- Integration test: create a small project with manifest + generator, verify it compiles and `EnsureModelAsync` is available

## Acceptance Criteria

- [ ] Source generator reads `model-manifest.json` as `AdditionalFiles`
- [ ] Generates a `partial static` class with `EnsureModelAsync`, `GetModelInfoAsync`, `VerifyModelAsync`, `ClearCache`
- [ ] Class name is customizable
- [ ] Generated class is `partial` so authors can extend it
- [ ] Works in Visual Studio (IntelliSense sees generated members)

### PR DESCRIPTION
**Pull request by luisquintanilla**
Closes #8

## Summary
Creates an `IIncrementalGenerator` that reads `model-manifest.json` at compile time and generates a partial static wrapper class, reducing model package authoring to just a manifest file and a `.csproj`.

## New project: `src/ModelPackages.Generators/`
- **ModelPackageGenerator.cs**: incremental source generator triggered by `model-manifest.json` in `AdditionalFiles`
- **ManifestParser.cs**: lightweight JSON parser for extracting model ID (no STJ dependency, netstandard2.0 compatible)
- **Diagnostics.cs**: `MPKG001` (empty manifest) and `MPKG002` (malformed manifest)

## Generated class includes
- `EnsureModelAsync`, `EnsureFilesAsync`, `GetModelInfoAsync`, `VerifyModelAsync`, `ClearCache`
- All delegate to a `Lazy<ModelPackage>` backed by embedded resource
- Class is `partial` so authors can extend with inference-specific methods
- Class name customizable via `ModelPackageClassName` MSBuild property

## Dependency note
Depends on #2 and #5 for stable API surface (the generated code delegates to these methods).

### CODE REVIEW

File: src/ModelPackages.Generators/ModelPackages.Generators.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <LangVersion>latest</LangVersion>
+    <Nullable>enable</Nullable>
+    <EnforceExtendedAnalyzerRules>true</EnforceExtendedAnalyzerRules>
+    <IsRoslynComponent>true</IsRoslynComponent>
+    <NoWarn>RS2008</NoWarn>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.*" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.*" PrivateAssets="all" />
```
[the remark this answers is not in the document]
Comment (luisquintanilla) [reply src/ModelPackages.Generators/ModelPackages.Generators.csproj]: **Rejected.** Floating versions are a deliberate choice for this prototype repository, consistent with the test project (PR #19). Pinning Roslyn SDK versions is appropriate for published analyzers but overly prescriptive for a prototype generator that targets the latest SDK.
